packages/main/src/plugin/updater.spec.ts

```diff
@@ -0,0 +1,156 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { beforeEach, expect, test, vi } from 'vitest';
+import type { MessageBox } from '/@/plugin/message-box.js';
+import type { StatusBarRegistry } from '/@/plugin/statusbar/statusbar-registry.js';
+import type { CommandRegistry } from '/@/plugin/command-registry.js';
+import { Disposable } from '/@/plugin/types/disposable.js';
+import { UPDATER_UPDATE_AVAILABLE_ICON } from '/@/plugin/index.js';
+import { Updater } from '/@/plugin/updater.js';
+import { app } from 'electron';
+import { type AppUpdater, autoUpdater } from 'electron-updater';
+import * as util from '/@/util.js';
+import type { AppUpdaterEvents } from 'electron-updater/out/AppUpdater.js';
+
+vi.mock('electron', () => ({
+  app: {
+    getVersion: vi.fn(),
+  },
+}));
+
+vi.mock('electron-updater', () => ({
+  autoUpdater: {
+    downloadUpdate: vi.fn(),
+    quitAndInstall: vi.fn(),
+    checkForUpdates: vi.fn(),
+    on: vi.fn(),
+    autoDownload: true,
+  },
+}));
+
+vi.mock('/@/util.js', () => ({
+  isLinux: vi.fn(),
+}));
+
+const messageBoxMock = {
+  showMessageBox: vi.fn(),
+} as unknown as MessageBox;
+
+const statusBarRegistryMock = {
+  setEntry: vi.fn(),
+} as unknown as StatusBarRegistry;
+
+const commandRegistry = {
+  registerCommand: vi.fn(),
+  executeCommand: vi.fn(),
+} as unknown as CommandRegistry;
+
+beforeEach(() => {
+  vi.useFakeTimers();
+  vi.resetAllMocks();
+
+  // Simulate PROD env
+  vi.stubEnv('PROD', 'true');
+
+  vi.spyOn(app, 'getVersion').mockReturnValue('@debug');
+  // eslint-disable-next-line no-null/no-null
+  vi.spyOn(autoUpdater, 'checkForUpdates').mockResolvedValue(null);
+
+  vi.spyOn(commandRegistry, 'executeCommand').mockResolvedValue(undefined);
+  vi.spyOn(util, 'isLinux').mockReturnValue(false);
```

Comment:
note: as it's mocked you can avoid the spy and do vi.mocked(app.getVersion).mockReturnValue('@debug');

Reply:
done